### ISSUE TITLE
Package request: kulala-ls

### ISSUE THREAD

**Issue by wenjinnn** [NixOS/nixpkgs#347263]
**Project description**
A minimal language server for HTTP syntax.

**Metadata**

* homepage URL: https://github.com/mistweaverco/kulala-ls
* source URL: https://github.com/mistweaverco/kulala-ls
* license: mit
* platforms: unix, linux, darwin, windows

---

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

**Some thoughts**
Maybe it should classified to nodePackages?

**Comment by dtomvan** [comment]
Found a solution in #385105. Could you review?

**Comment by wenjinnn** [comment]
> Found a solution in [#385105](https://github.com/NixOS/nixpkgs/pull/385105). Could you review?

LGTM! just tested that out and it worked for me, but unfortunately I don't have the permission to approve it.
FYI, for neovim user, if you want to use kulala-ls before this PR is merged, you can modify the neovim kulala-ls cmd to start it from npx like this, it's a workaround that I'm using:
```lua
      require("lspconfig").kulala_ls.setup({
        cmd = { "npx", "--yes", "--", "@mistweaverco/kulala-ls", "--stdio" },
      })
```

> There's just one problem. When `npm run build` gets run, in a normal environment it **downloads and runs** `tree-sitter-cli`**. We should probably provide a link to `lib.getExe pkgs.tree-sitter` but I do not know how to integrate that into the build process...

I've been stuck here before, and I give it up and using the npx way. Great work!

**Comment by dtomvan** [comment]
> but unfortunately I don't have the permission to approve it. 

Yeah but I need to at least prove to them that `kulala-ls` is a package that is not too niche and should be included in nixpkgs. So I wanted to ask these people who requested these packages to drop their LGTM's 😆 . So thanks for that.

> I've been stuck here before
What I did is include a patch that removes `tree-sitter-cli` from both `package.json`'s that refer to it. Then `tree-sitter` does not get pulled during build time by `install.js`, because it's not included. The `pkg/tree-sitter` grammar tree does specify a build script of `tree-sitter build` but by including `pkgs.tree-sitter` in `nativeBuildInputs` the nix version of tree-sitter can be picked up by npm.

Fun fact: `git clone https://github.com/mistweaverco/kulala-ls && cd kulala-ls && npm i && npm run build` does NOT work on NixOS outside of a build, because the `tree-sitter` binary that is pulled needs to be patched to link to /nix/store entries.
